07-routing-gossip.md

```diff
@@ -0,0 +1,252 @@
+
+The creating node MUST set `timestamp` to greater than zero, and MUST set it to greater than any previously-sent `channel_update` for this channel, and MUST set `pad` to zero.
+
+It MUST set `expiry` to the number of blocks it will subtract from an incoming HTLC's `expiry`.  It MUST set `htlc-minimum-msat` to the minimum HTLC value it will accept, in millisatoshi.  It MUST set `fee-base-msat` to the base fee it will charge for any HTLC, in millisatoshi, and `fee-proportional-millionths` to the amount it will charge per millionth of a satoshi.
+
+The node SHOULD accept HTLCs which pay a fee equal or greater than:
+
+	fee-base-msat + htlc-amount-msat * fee-proportional-millionths / 1000000
+
+The receiving node SHOULD fail the connection if `side` is not 0 or 1.
+It MUST ignore the contents of `pad`.  The receiving node SHOULD fail
+the connection if `signature` is invalid or incorrect for the entire
+message including unknown fields following `signature`, and MUST NOT
+further process the message.  The receiving node SHOULD ignore `ipv6`
+if `port` is zero.  It SHOULD ignore the message if `blockheight`,
+`blockindex` and `outputindex` do not correspond to a previously
+known, unspent channel from `channel_announcement` or if `timestamp`
+is not greater than than the last-received `channel_announcement` for
+this channel and `side`.  Otherwise, if the `timestamp` is equal to
+the last-received `channel_announcement` and the fields other than
+`signature` differ, the node MAY blacklist this node-id and forget all
+channels associated with it.  Otherwise the receiving node SHOULD
+queue the message for rebroadcasting.
+
+## Rebroadcasting
+
+Nodes receiving an announcement verify that this is the latest update from the announcing node by comparing the included timestamp and update their local view of the network's topology accordingly.
+
+Once the announcement has been processed it is added to a list of outgoing announcements to the processing node's peers, which will be flushed at regular intervals.
+This store and delayed forward broadcast is called a _staggered broadcast_
+
+If, after applying the changes from the announcement, there are no channels associated with the announcing node, then the receiving node MAY purge the announcing node from the set of known nodes.
+Otherwise the receiving node updates the metadata and stores the signature associated with the announcement.
+This will later allow the receiving node to rebuild the announcement for its peers.
+
+After processing the announcement the receiving node adds the announcement to a list of outgoing announcements.
+
+### Requirements
+
+Each node SHOULD flush outgoing announcements once every 60 seconds, independently of the arrival times of announcements, resulting in a staggered announcement and deduplication of announcements.
+
+Nodes MAY re-announce their channels regularly, however this is discouraged in order to keep the resource requirements low.
```

Comment:
Should we add some kind of blacklisting based on a max number of `node_announcement` or `channel_update` that a peer is allowed to send per second, considering that those messages are costless to produce?